src/waveform/renderers/qopengl/shaders/endoftrackshader.cpp

```diff
@@ -0,0 +1,29 @@
+#include "waveform/renderers/qopengl/shaders/endoftrackshader.h"
+
+using namespace qopengl;
+
+void EndOfTrackShader::init() {
+    QString vertexShaderCode = QStringLiteral(R"--(
+attribute vec4 position;
+attribute float gradient;
+varying float vgradient;
+void main()
+{
+    vgradient = gradient;
+    gl_Position = position;
+}
```

Comment:
isn't this missing the matrix?

Reply:
no need for a matrix, as the end of track applies to the entire view (so it just works on the coordinates going from 0,0 to 1,1)

Reply:
Ok, I didn't know that it was automatically transformed into the view-space.